magefile.go

```diff
@@ -1250,54 +1256,201 @@ func (d Dev) deployAPIMock(ctx context.Context, cluster *dev.Cluster) error {
+		TestAddonsPlugAndPlayHandler{},
+		TestMonitoringStackHandler{},
+	}
+	return handler
+}
+
+func (hl TestHandlerList) ReconcileFlags(ctx context.Context, client ctrlclient.Client) error {
+	for _, h := range hl {
+		if err := h.ReconcileFlags(ctx, client); err != nil {
+			return err
 		}
 	}
 	return nil
 }
 
-func preClusterCreationFeatureToggleSetup(ctx context.Context) error {
-	availableFeatureToggles := featuretoggle.GetAvailableFeatureToggles()
+func (hl TestHandlerList) postClusterCreationFeatureFlagSetup(ctx context.Context, cluster *dev.Cluster) error {
+	for _, h := range hl {
+		if err := h.postClusterCreationFeatureFlagSetup(ctx, cluster); err != nil {
+			return err
+		}
+	}
+	return nil
+}
 
-	for _, featTog := range availableFeatureToggles {
-		// feature toggles enabled/disabled at the level of openshift/release in the form of multiple jobs
-		if featuretoggle.IsEnabledOnTestEnv(featTog) {
-			if err := featTog.PreClusterCreationSetup(ctx); err != nil {
-				return fmt.Errorf("failed to set the feature toggle before the cluster creation: %w", err)
-			}
+type TestAddonsPlugAndPlayHandler struct{}
+
+func (h TestAddonsPlugAndPlayHandler) ReconcileFlags(ctx context.Context, client ctrlclient.Client) error {
+	if testutil.IsEnabledOnTestEnv(featureflag.AddonsPlugAndPlayFeatureFlagIdentifier) {
+		return EnableFeatureFlag(ctx, client, featureflag.AddonsPlugAndPlayFeatureFlagIdentifier)
+	}
+	return DisableFeatureFlag(ctx, client, featureflag.AddonsPlugAndPlayFeatureFlagIdentifier)
+}
+
+func (h TestAddonsPlugAndPlayHandler) postClusterCreationFeatureFlagSetup(ctx context.Context, cluster *dev.Cluster) error {
+	if testutil.IsEnabledOnTestEnv(featureflag.AddonsPlugAndPlayFeatureFlagIdentifier) {
+		if err := deployPackageOperator(ctx, cluster); err != nil {
+			return fmt.Errorf("failed to deploy pko: %w", err)
 		}
 	}
 	return nil
 }
 
-func postClusterCreationFeatureToggleSetup(ctx context.Context, cluster *dev.Cluster) error {
-	availableFeatureToggles := featuretoggle.GetAvailableFeatureToggles(
-		featuretoggle.WithClient{Client: cluster.CtrlClient},
-		featuretoggle.WithSchemeToUpdate{Scheme: cluster.Scheme},
-	)
+type TestMonitoringStackHandler struct{}
+
+func (h TestMonitoringStackHandler) ReconcileFlags(ctx context.Context, client ctrlclient.Client) error {
+	if testutil.IsEnabledOnTestEnv(featureflag.MonitoringStackFeatureFlagIdentifier) {
+		return EnableFeatureFlag(ctx, client, featureflag.MonitoringStackFeatureFlagIdentifier)
+	}
+	return DisableFeatureFlag(ctx, client, featureflag.MonitoringStackFeatureFlagIdentifier)
+}
+
+func (h TestMonitoringStackHandler) postClusterCreationFeatureFlagSetup(ctx context.Context, cluster *dev.Cluster) error {
+	if testutil.IsEnabledOnTestEnv(featureflag.MonitoringStackFeatureFlagIdentifier) {
+		if err := deployObservabilityOperator(ctx, cluster); err != nil {
+			return fmt.Errorf("failed to deploy observability operator: %w", err)
+		}
+	}
+	return nil
+}
```

Comment:
I think I suggested this [on Slack](https://redhat-internal.slack.com/archives/D03EWADJH47/p1682456516811499) instead of a Github comment here. So let me do so. 
Bringing the "testable"-interface of featureToggles under Magefile would only pollute it much further and this pattern would subject it to get polluted even further in the future.

PS: I am also not personally happy with the current state of the magefile, it's very hard to navigate too cluttered up at the moment due to the absence of a packaging convention in it. 

Normally, as a developer and reader of the code, I would expect everything around featuretoggles, including testable-methods such as PostClusterCreationSetup/Enable/Disable to be under its own package for better visiblity and at the same time, making it convenient for the dev/reader going through magefile to be safe from all the clutter introduced all the more by the feature-toggle oriented stuff in it.

wdyt :)